lib/Ast.ml

```diff
@@ -203,6 +210,25 @@ module Longident = struct
   let is_index_op i = Longident.last i |> String_id.is_index_op
 
   let is_symbol i = is_prefix i || is_infix i || is_index_op i
+
+  (** [fit_margin c x] returns [true] if and only if [x] does not exceed 2/3
+      of the margin. *)
+  let fit_margin (c : Conf.t) x = x * 3 < c.margin * 2
+
+  let is_simple c x =
+    let rec length (x : Longident.t) =
+      match x with
+      | Lident x -> String.length x
+      | Ldot (x, y) -> length x + 1 + String.length y
+      | Lapply (x, y) -> length x + length y + 3
+    in
+    fit_margin c (length x)
+end
+
+module Attr = struct
+  let is_doc = function
+    | {attr_name= {Location.txt= "ocaml.doc" | "ocaml.text"; _}; _} -> false
+    | _ -> true
```

Comment:
You meant to name this `is_not_doc` ?

Reply:
Unfortunately it's already present in the original version like this :/

I'd rather not have negatives in function names, so I suggest either naming it `is_normal_attribute` (`is_attribute` / `has_attribute_syntax` maybe?) or keeping `is_doc_comment`, inverting the booleans and fixing the call sites (`List.for_all ~f:is_doc_comment xxx_attributes`).

Sidenote, I remember extracting this function in `Cmts.ml` so you can remove it and convert it to use this one if applicable.

Reply:
I can't believe we missed this one all this time, fixed thanks!